### PR TITLE
Remove VMS, part 1

### PR DESCRIPTION
**Pull request by richsalz**
This is part one (of two, I think) of removing VMS from the source tree.

It removes __VMS code from the source.
Removes "are we running on VMS?" from the testss
Removes VMS-specific files.
Remove build notes/docs about running on VMS.
- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

File: Configure [hunk 1]
```diff
@@ -744,7 +736,7 @@ my %user = (
     LDLIBS      => [ env('LDLIBS') || () ],  # -l
     MT          => undef,
     MTFLAGS     => [],
-    PERL        => env('PERL') || ($^O ne "VMS" ? $^X : "perl"),
+    PERL        => env('PERL') || "perl",
```
Comment (botovq) [inline Configure:739]:
```suggestion
    PERL        => env('PERL') || $^X,
```

File: Configure [hunk 2]
```diff
@@ -3482,18 +3458,11 @@ sub print_table_entry
...
     # On non-platforms, we just use file_name_is_absolute().
-    return file_name_is_absolute($file) unless $^O eq "VMS";
-
-    # If the file spec includes a device or a directory spec,
-    # file_name_is_absolute() is perfectly safe.
-    return file_name_is_absolute($file) if $file =~ m|[:\[]|;
+    return file_name_is_absolute($file);
 
     # Here, we know the given file spec isn't absolute
     return 0;
```
Comment (botovq) [inline Configure:3468]: The three previous lines are unreachable.

File: apps/rehash.c [hunk 3]
```diff
@@ -327,10 +320,7 @@ static int ends_with_dirsep(const char *path)
 {
     if (*path != '\0')
         path += strlen(path) - 1;
-# if defined __VMS
-    if (*path == ']' || *path == '>' || *path == ':')
-        return 1;
-# elif defined _WIN32
+# if   defined _WIN32
```
Comment (botovq) [inline apps/rehash.c:323]:
```suggestion
# if defined _WIN32
```

File: configdata.pm.in [hunk 4]
```diff
@@ -15,21 +15,6 @@
      # Make the path absolute at all times
      $path = abs_path($path);
```
Comment (botovq) [inline configdata.pm.in:16]: Can this be simplified to `return abs_path($path);`? Can we drop `_fixup_path` entirely and inline `abs_path()` in the two callers directly?

File: crypto/LPdir_unix.c [hunk 5]
```diff
@@ -100,14 +93,6 @@ const char *LP_find_file(LP_DIR_CTX **ctx, const char *directory)
         }
         memset(*ctx, 0, sizeof(**ctx));
```
Comment (botovq) [inline crypto/LPdir_unix.c:95]: zap extra empty line

File: crypto/LPdir_unix.c [hunk 6]
```diff
@@ -119,12 +104,6 @@ const char *LP_find_file(LP_DIR_CTX **ctx, const char *directory)
         }
     }
```
Comment (botovq) [inline crypto/LPdir_unix.c:106]: ditto

File: crypto/bio/bio_sock.c [hunk 7]
```diff
@@ -33,9 +33,7 @@ static int wsa_init_done = 0;
 #  include <winsock.h> /* for type fd_set */
 # else
 #  include <unistd.h>
-#  if defined __VMS
-#   include <sys/socket.h>
-#  elif defined _HPUX_SOURCE
+#  if   defined _HPUX_SOURCE
```
Comment (botovq) [inline crypto/bio/bio_sock.c:36]:
```suggestion
#  if defined _HPUX_SOURCE
```

File: fuzz/test-corpus.c [hunk 8]
```diff
@@ -80,9 +80,6 @@ int main(int argc, char **argv) {
                 if (pathname == NULL)
                     break;
                 strcpy(pathname, argv[n]);
-#ifdef __VMS
-                if (strchr(":<]", pathname[dirname_len - 1]) == NULL)
-#endif
                     pathname[dirname_len++] = '/';
```
Comment (botovq) [inline fuzz/test-corpus.c:83]:
```suggestion
                pathname[dirname_len++] = '/';
```

File: include/internal/common.h [hunk 9]
```diff
@@ -188,10 +188,7 @@ static ossl_inline int ossl_ends_with_dirsep(const char *path)
 {
     if (*path != '\0')
         path += strlen(path) - 1;
-# if defined __VMS
-    if (*path == ']' || *path == '>' || *path == ':')
-        return 1;
-# elif defined _WIN32
+# if   defined _WIN32
```
Comment (botovq) [inline include/internal/common.h:191]:
```suggestion
# if defined _WIN32
```

File: include/internal/common.h [hunk 10]
```diff
@@ -205,22 +202,14 @@ static ossl_inline char ossl_determine_dirsep(const char *path)
...
 static ossl_inline int ossl_is_absolute_path(const char *path)
 {
-# if defined __VMS
-    if (strchr(path, ':') != NULL
-        || ((path[0] == '[' || path[0] == '<')
-            && path[1] != '.' && path[1] != '-'
-            && path[1] != ']' && path[1] != '>'))
-        return 1;
-# elif defined _WIN32
+# if   defined _WIN32
```
Comment (botovq) [inline include/internal/common.h:212]:
```suggestion
# if defined _WIN32
```

File: include/internal/e_os.h [hunk 11]
```diff
@@ -189,7 +189,6 @@ FILE *__iob_func(void);
 #  endif
 
 #  ifdef OPENSSL_SYS_VMS
```
Comment (botovq) [inline include/internal/e_os.h:191]: Can't this all go away until the corresponding `#endif`?

File: providers/implementations/storemgmt/file_store.c [hunk 12]
```diff
@@ -611,15 +611,6 @@ static int file_name_check(struct file_ctx_st *ctx, const char *name)
...
 
-#ifdef __VMS
-    /*
-     * One extra step here, check for a possible generation number.
-     */
-    if (*p == ';')
-        for (p++; *p != '\0'; p++)
-            if (!ossl_isdigit((unsigned char)*p))
-                break;
-#endif
```
Comment (botovq) [inline providers/implementations/storemgmt/file_store.c:614]: Extra blank line

File: test/recipes/02-test_ordinals.t [hunk 13]
```diff
@@ -38,7 +38,7 @@ sub testordinals
             #Ignore this line
             next;
         }
-        #Some ordinals can be repeated, e.g. if one is VMS and another is !VMS
+        #Some ordinals can be repeated, e.g. if on different platforms
```
Comment (botovq) [inline test/recipes/02-test_ordinals.t:41]: I don't understand this comment (before and after the edit).

File: test/recipes/80-test_pkcs12.t [hunk 14]
```diff
@@ -40,17 +40,6 @@ if (eval { require Win32::API; 1; }) {
     }
 } elsif ($^O eq "MSWin32") {
     plan skip_all => "Win32::API unavailable";
-} elsif ($^O ne "VMS") {
```
Comment (botovq) [inline test/recipes/80-test_pkcs12.t:43]: This block should probably stay?
```diff
-} elsif ($^O ne "VMS") {
+} else {
```

File: test/recipes/82-test_ocsp_cert_chain.t [hunk 15]
```diff
@@ -25,8 +25,8 @@ plan skip_all => "$test_name requires sock enabled"
     if disabled("sock");
 plan skip_all => "$test_name requires TLS enabled"
     if alldisabled(available_protocols("tls"));
-plan skip_all => "$test_name is not available Windows or VMS"
-    if $^O =~ /^(VMS|MSWin32|msys)$/;
+plan skip_all => "$test_name is not available Windows"
```
Comment (botovq) [inline test/recipes/82-test_ocsp_cert_chain.t:28]:
```suggestion
plan skip_all => "$test_name is not available on Windows"
```

File: util/mkdef.pl [hunk 16]
```diff
@@ -44,7 +43,6 @@
            'type=s'     => \$type,
            'ctest'      => \$ctest,
            'verbose'    => \$verbose,
-           # For VMS
            'case-insensitive' => \$case_insensitive)
```
Comment (botovq) [inline util/mkdef.pl:46]: Can't this go since it was only for VMS?

File: util/perl/OpenSSL/Test.pm [hunk 17]
```diff
@@ -435,9 +435,6 @@ sub run {
     return () if !$cmd;
 
     my $prefix = "";
```
Comment (botovq) [inline util/perl/OpenSSL/Test.pm:437]: I think `$prefix` can go entirely, since it's always the empty string.

File: util/perl/OpenSSL/Util.pm [hunk 18]
```diff
@@ -151,17 +150,7 @@ sub fixup_cmd_elements {
...
-            $_ = shift;
-            if ($_ eq '' || /\s|[!"[:upper:]]/) {
-                s/"/""/g;
-                '"'.$_.'"';
-            } else {
-                $_;
-            }
-        };
-    } elsif ( $^O eq "MSWin32") { # MSWin setup
+    if ( $^O eq "MSWin32") { # MSWin setup
```
Comment (botovq) [inline util/perl/OpenSSL/Util.pm:153]:
```suggestion
    if ( $^O eq "MSWin32") {
```

File: util/perl/OpenSSL/Util.pm [hunk 19]
```diff
@@ -182,57 +171,13 @@ sub fixup_cmd_elements {
 
 This is a sibling of fixup_cmd_elements() that expects the LIST to be a
 complete command line.  It does the same thing as fixup_cmd_elements(),
```
Comment (botovq) [inline util/perl/OpenSSL/Util.pm:173]:
```suggestion
complete command line.
```
Or at least change the comma to a period.

File: util/wrap.pl.in [hunk 20]
```diff
@@ -47,29 +47,13 @@ local $ENV{OPENSSL_MODULES} = $std_providers
 local $ENV{OPENSSL_CONF} = $std_openssl_conf
     if ($ENV{OPENSSL_CONF} // '') eq '' && -f $std_openssl_conf;
 {-
-     # For VMS, we define logical names to get the libraries properly
-     # defined.
      use File::Spec::Functions qw(rel2abs);
```
Comment (botovq) [inline util/wrap.pl.in:50]: Is this still needed or can the entire thing enclosed in `{-` and `-}` go away?

**Review by botovq — COMMENTED** [review]
Made a first pass. I assume I haven't caught all that is relevant to the CI breakage which I haven't looked at, yet.

File: util/wrap.pl.in (hunk 20 above)
Comment (richsalz) [reply util/wrap.pl.in:50]: the entire thing can go away.

**Comment by richsalz** [comment]
Added a commit to fix the things you found so far

**Comment by botovq** [comment]
This one will require another pass with a fresh head, so I'll look again tomorrow. There's also conflicts that will need to be resolved.

**Comment by richsalz** [comment]
Rebased, fixed conflicts, clean slate next time you look at it :)

**Comment by botovq** [comment]
Would you mind rebasing this so we can see if there's other fallout than the leading minuses in sockets.h?

**Comment by richsalz** [comment]
Rebased and pushed.

**Review by botovq — COMMENTED** [review]
It feels like the failures are a consequence of this PR, although I can't spot the root cause.Feels like some #ifdef went wrong for windows 32 vs 64 bit.

**Comment by richsalz** [comment]
It's always the same test, but different cases sometimes, that fail:
```
80-test_ssl_new.t                     (Wstat: 2560 Tests: 31 Failed: 10)
  Failed tests:  2, 4, 7, 10-11, 18, 20, 25, 27-28
```

**Comment by botovq** [comment]
Is this still broken or did some recent changes resolve the problems?

**Comment by richsalz** [comment]
Tests are still failing, all in ssl_old.  Still broken.  STill need to take a look :(

**Comment by richsalz** [comment]
I am restarting this PR over.  Closing this and starting a new PR.

**Pull request closed without merging**